src/notice/notice_backtrace.rs

```diff
@@ -0,0 +1,141 @@
+use crate::backtrace::{Backtrace, BacktraceFrame, BacktraceSymbol};
+use serde::ser::{Serialize, SerializeSeq, Serializer};
+use serde_json::{self, Value};
+use std::collections::HashMap;
+
+#[derive(Debug, Clone)]
+pub struct NoticeTrace {
+    frames: Vec<NoticeFrame>,
+}
+
+impl NoticeTrace {
+    fn new(frames: Vec<NoticeFrame>) -> NoticeTrace {
+        NoticeTrace { frames }
+    }
+
+    pub fn frames(&self) -> Vec<NoticeFrame> {
```

Comment:
The clone here isn't great- is there a specific reason it's cloned? This probably warrants a couple comments either way

Reply:
Resolved this by completely removing the NoticeTrace struct, since it was somewhat redundant